File: ModelDtos/Users/ChangePasswordProfileRequest.cs
using System.ComponentModel.DataAnnotations;

namespace _24hplusdotnetcore.ModelDtos.Users
{
    public class ChangePasswordProfileRequest
    {
        [Required(ErrorMessage = "{0} is required")]
        public string NewPassword { get; set; }

        [Required(ErrorMessage = "{0} is required")]
        public string OldPassword { get; set; }
    }
}

File: src/modules/profile/dto/change-password-profile.request.ts
import { IsNotEmpty, IsString, MinLength } from 'class-validator';

export class ChangePasswordProfileRequest {
  @IsString()
  @IsNotEmpty()
  old_password: string;

  @IsString()
  @IsNotEmpty()
  @MinLength(6)
  new_password: string;
}

